packages/node/src/index.ts

```diff
@@ -201,18 +202,43 @@ async function compile(
         if (typeof cached !== 'undefined') return cached;
 
         try {
+          let entry: File | undefined;
           let source: string | Buffer = readFileSync(fsPath);
+
+          const { mode } = lstatSync(fsPath);
+          if (isSymbolicLink(mode)) {
+            entry = new FileFsRef({ fsPath, mode });
+          }
+
+          if (isEdgeFunction && basename(fsPath) === 'package.json') {
+            // For Edge Functions, patch "main" field to prefer "browser" or "module"
+            const pkgJson = JSON.parse(source.toString());
+            for (const prop of ['browser', 'module']) {
+              const val = pkgJson[prop];
+              if (typeof val === 'string') {
+                debug(`Using "${prop}" field in ${fsPath}`);
+                pkgJson.main = val;
+
+                // Create the `entry` with the original so that the output is unmodified
+                if (!entry) {
+                  entry = new FileBlob({ data: source, mode });
+                }
```

Comment:
Do we need to create the `entry` here? It seems like it would be created below on line 241 regardless of what happens here.

Reply:
Agreed! In fact, I think you can move the `if` at line 213 to an `else if` following line 239

Reply:
`source` gets mutated on the next line though, so we need to add the entry here so that it contains the original `package.json` contents.

Reply:
agreed